scripts/dts/gen_defines.py

```diff
@@ -707,7 +707,7 @@ def out_node(node, ident, val, name_alias=None):
     # Returns the identifier used for the macro that provides the value
     # for 'ident' within 'node', e.g. DT_MFG_MODEL_CTL_GPIOS_PIN.
 
-    node_prefix = node_ident(node)
+    node_prefix = "PATH_" + str2ident(node.path[1:])
```

Comment:
Could do

```
# /foo/bar -> PATH_FOO_BAR
node_prefix = "PATH" + str2ident(node.path)
```

to take advantage of the initial `/` turning into `_`.

Reply:
Wish the scripts would've started out with `__` or the like between the "parts" in identifiers. It's hard to tell where they start and end now. Stuff often bleeds together.

Unrelated though.

Reply:
In C++ identifiers that contain a double underscore anywhere are reserved.  Macro names are identifiers.  So as long as the header must be processed by C++ it can't have double underscores in the names.

I'm aware Linux uses `__`.  Zephyr is not Linux, and certainly toolchain portability that's desired for Zephyr mandates more strict conformance to language specifications than Linux might require.